ConfigSpace/hyperparameters.pyx

```diff
@@ -519,7 +517,7 @@ cdef class UniformFloatHyperparameter(FloatHyperparameter):
         return np.maximum(self.lower, np.minimum(self.upper, vector))
 
     cpdef double _transform_scalar(self, double scalar):
-        if not math.isfinite(scalar):
+        if scalar == np.nan:
```

Comment:
Unfortunately, that won't work as comparing to `np.NaN` always results in False:
```                                                                                                                                                          
In [3]: a == np.NaN                                                                                                                                                            
Out[3]: False
In [4]: a = 5                                                                                                                                                                  
In [5]: a == np.NaN                                                                                                                                                            
Out[5]: False
```

Reply:
Instead, you can do:
```
In [8]: a = np.NaN                                                                                                                                                             
In [9]: a != a                                                                                                                                                                 
Out[9]: True
```